frontend/ui/screensaver.lua

```diff
@@ -60,93 +37,204 @@ local function getRandomImage(dir)
     if i == 0 then
         return nil
     end
-    return createWidgetFromFile(dir .. pics[math.random(i)])
+    return dir .. pics[math.random(i)]
 end
 
-function Screensaver:isUsingBookCover()
-    -- this setting is on by default
-    return G_reader_settings:readSetting('use_lastfile_as_screensaver') ~= false
+function Screensaver:chooseFolder()
+    local buttons = {}
+    table.insert(buttons, {
+        {
+            text = _("Choose screensaver directory by long-pressing"),
+            callback = function()
+                UIManager:close(self.choose_dialog)
+                require("ui/downloadmgr"):new{
+                    title = _("Choose screensaver directory"),
+                    onConfirm = function(path)
+                        logger.dbg("set screensaver directory to", path)
+                        G_reader_settings:saveSetting("screensaver_dir", path)
+                        UIManager:show(InfoMessage:new{
+                            text = T(_("Screensaver directory set to:\n%1"), path),
+                            timeout = 3,
+                        })
+                    end,
+                }:chooseDir()
+            end,
+        }
+    })
+    table.insert(buttons, {
+        {
+            text = _("Close"),
+            callback = function()
+                UIManager:close(self.choose_dialog)
+            end,
+        }
+    })
+    local screensaver_dir = G_reader_settings:readSetting("screensaver_dir")
```

Comment:
My previous random image directory setting was not re-used (`["suspend_screensaver"]["path"]`).
Just mentionning it, not sure it really needs to be migrated from this old setting.

Reply:
The old settings were too complicated so they were not migrated :)